test/e2e/providers/exec/ssh.go

```diff
@@ -44,7 +44,7 @@ func RunSSHCommand(host string, config *ssh.ClientConfig, cmd string) (code int,
 			return 0, "", "", fmt.Errorf("did not get an exit status for SSH command: %v", e)
 		case *ssh.ExitError:
 			// SSH operation successful, but command returned error code
-			return e.ExitStatus(), stdoutB.String(), stderrB.String(), nil
+			return e.ExitStatus(), stdoutB.String(), stderrB.String(), err
```

Comment:
I think it returned nil as error on purpose according to the comment, could you explain what problem there is without the patch? I see client side has checked return code.